### PR TITLE
MBS-11112: Convert /main error pages to React

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11112

See each commit for details.

### CODE REVIEW

**Comment by mwiencek** [comment]
Converting the 500 error page should probably wait for https://github.com/metabrainz/musicbrainz-server/pull/1739 since even the .tt version barely works at all at present.

**Comment by reosarevok** [comment]
AFAICT this is now ready.

File: root/main/error/components/ErrorEnvironment.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import * as React from 'react';
+
+type Props = {
+  +$c: CatalystContextT,
+  +hostname?: string,
+  +useLanguages?: boolean,
+};
+
+const ErrorEnvironment = ({
```
Comment (yvanzo) [inline root/main/error/components/ErrorEnvironment.js:18]: Commit https://github.com/metabrainz/musicbrainz-server/pull/1699/commits/faf46ccac63eaa0d97b71c59dc05a63cdcb418e3 gives no indication of what this is being converted from.

File: root/main/error/components/ErrorInfo.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+import * as React from 'react';
+
+type Props = {
+  +formattedErrors?: $ReadOnlyArray<string>,
+  +message?: string,
+};
+
+const ErrorInfo = ({
```
Comment (yvanzo) [inline root/main/error/components/ErrorInfo.js:17]: Same here, I just found it comes from `/root/main/info/error.tt` but commit https://github.com/metabrainz/musicbrainz-server/pull/1699/commits/b5ef9f12c395ef4135313c62470437d5fdb48c08 gave no indication.

**Review by yvanzo — CHANGES_REQUESTED** [review]
Please clarify the origin of converted files in commit messages, I did not review them all.

File: root/main/error/components/ErrorEnvironment.js (hunk 1 above)
Comment (reosarevok) [reply root/main/error/components/ErrorEnvironment.js:18]: Fair enough, added the info to the commit message

File: root/main/error/500.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    <ErrorEnvironment
+      $c={$c}
+      hostname={hostname}
+      useLanguages={useLanguages}
+    />
+
+    <p>
+      {l(`We’re terribly sorry for this problem.
+          Please wait a few minutes and repeat
+          your request — the problem may go away.`)}
```
Comment (yvanzo) [inline root/main/error/500.js:79]: This will obsolete existing translations for the previous message.

File: root/main/error/500.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+type Props = {
+  +$c: CatalystContextT,
+  +edits?: $ReadOnlyArray<EditT>,
+  +formattedErrors?: $ReadOnlyArray<string>,
+  +hostname?: string,
+  +useLanguages: boolean,
+};
+
+const Error500 = ({
```
Comment (yvanzo) [inline root/main/error/500.js:28]: Shouldn’t filename match component’s name like we did so far? That is `Error500.js` instead of `500.js`.

File: root/main/error/components/ErrorEnvironment.js (hunk 1 above)
Comment (yvanzo) [reply root/main/error/components/ErrorEnvironment.js:18]: Thank you!

File: root/main/error/500.js (hunk 4 above)
Comment (reosarevok) [reply root/main/error/500.js:28]: I just followed what we did for 404, but I'd prefer this too. I'll change it in an extra commit.

File: root/main/error/500.js (hunk 3 above)
Comment (reosarevok) [reply root/main/error/500.js:79]: React would otherwise just print the &#x2014; unchanged. I guess we could avoid that by using expand in some way, but maybe it's not such a big issue?

Comment (yvanzo) [reply root/main/error/500.js]: That’s why, thanks. No it isn’t a big issue.

**Review by yvanzo — APPROVED** [review]
Thanks!

File: lib/MusicBrainz/Server/Controller/Root.pm [hunk 5]
```diff
@@ -166,7 +166,10 @@ sub error_404 : Private {
 
     $c->response->status(404);
     $c->stash->{current_view} = 'Node';
-    $c->stash->{component_props}{message} = $message;
+    $c->stash->{component_props} = {
+        message => $message,
+        template => 'main/error/404',
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Controller/Root.pm]: This change seems unrelated to the commit and possibly a no-op (re the `template` addition), unless I'm missing something.

**Review by mwiencek — COMMENTED** [review]
This looks ready for merge, just wondering about the comment on `template => 'main/error/404',` below.

File: lib/MusicBrainz/Server/Controller/Root.pm (hunk 5 above)
Comment (reosarevok) [reply lib/MusicBrainz/Server/Controller/Root.pm]: I'm moving the file from main/404 to main/error/404, wouldn't that require specifying the template?

Comment (reosarevok) [reply lib/MusicBrainz/Server/Controller/Root.pm]: I was being an idiot, fixed that now.

**Review by mwiencek — APPROVED** [review]
